src/GitHub.Services.Vssdk/Services/TippingService.cs

```diff
@@ -0,0 +1,102 @@
+        }
+
+        /// <inheritdoc/>
+        public void RequestCalloutDisplay(Guid calloutId, string title, string message,
+            bool isPermanentlyDismissible, FrameworkElement targetElement,
+            Guid vsCommandGroupId, uint vsCommandId)
+        {
+            var screenPoint = !Splat.ModeDetector.InUnitTestRunner() ?
+                    targetElement.PointToScreen(new Point(targetElement.ActualWidth / 2, 0)) : default;
+            var point = new Microsoft.VisualStudio.OLE.Interop.POINT { x = (int)screenPoint.X, y = (int)screenPoint.Y };
+            RequestCalloutDisplay(ClientId, calloutId, title, message, isPermanentlyDismissible,
+                point, vsCommandGroupId, vsCommandId);
+        }
+
+        // The option to pass a command option is only available on Visual Studio 2017+.
+        void RequestCalloutDisplay(Guid clientId, Guid calloutId, string title, string message,
+            bool isPermanentlyDismissible, FrameworkElement targetElement,
+            Guid vsCommandGroupId, uint vsCommandId, object commandOption = null)
+        {
+            var tippingService = serviceProvider.GetService(typeof(SVsTippingService));
+            Assumes.Present(tippingService);
+            var parameterTypes = new Type[] { typeof(Guid), typeof(Guid), typeof(string), typeof(string),
+                typeof(bool), typeof(FrameworkElement), typeof(Guid), typeof(uint), typeof(object) };
+            var method = tippingService.GetType().GetInterfaces()
+                .Where(i => i.Name == "IVsTippingService")
+                .FirstOrDefault()?.GetMethod("RequestCalloutDisplay", parameterTypes);
+            var arguments = new object[] { clientId, calloutId, title, message, isPermanentlyDismissible, targetElement,
+                    vsCommandGroupId, vsCommandId, commandOption };
+            method.Invoke(tippingService, arguments);
+        }
+
+        // Available on Visual Studio 2015
+        void RequestCalloutDisplay(Guid clientId, Guid calloutId, string title, string message, bool isPermanentlyDismissible,
+            Microsoft.VisualStudio.OLE.Interop.POINT anchor, Guid vsCommandGroupId, uint vsCommandId)
+        {
+            var tippingService = serviceProvider.GetService(typeof(SVsTippingService));
+            if (tippingService == null)
+            {
+                log.Error("Can't find {ServiceType}", typeof(SVsTippingService));
+                return;
+            }
+
+            Assumes.Present(tippingService);
+            var parameterTypes = new Type[] { typeof(Guid), typeof(Guid), typeof(string), typeof(string), typeof(bool),
+                typeof(Microsoft.VisualStudio.OLE.Interop.POINT), typeof(Guid), typeof(uint) };
+            var tippingServiceType = tippingService.GetType();
+            var method = tippingService.GetType().GetInterfaces().Where(i => i.Name == "IVsTippingService")
+                .FirstOrDefault()?.GetMethod("RequestCalloutDisplay", parameterTypes);
```

Comment:
If you do `Type.GetType("Microsoft.Internal.VisualStudio.Shell.Interop.IVsTippingService, Microsoft.VisualStudio.Shell.UI.Internal, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a").GetMethod("RequestCalloutDisplay", ...)`, you can put it in a static readonly MethodInfo or delegate field.

You could also define `IVsTippingService` next to your `SVsTippingService` and do `typeof(IVsTippingService)`.

Reply:
Anyway; `Where(...).FirstOrDefault()` can be `FirstOrDefault(...)` and you can say `var parameterTypes = new[] {` if you like.

Reply:
I think I'd still do `BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly` ([REFL008](https://github.com/DotNetAnalyzers/ReflectionAnalyzers/blob/master/documentation/REFL008.md)).

Reply:
I tried defining my own `IVsTippingService`, but for some reason COM type equivalence wasn't working. 😕 

I've changed this method to look for the `IVsTippingService` type `GUID`. That should make sure it's unique, but avoids depending explicitly on the `Microsoft.VisualStudio.Shell.UI.Internal` assembly. The `IVsTippingService` interface might move in a future version.

> Anyway; Where(...).FirstOrDefault() can be FirstOrDefault(...) and you can say var parameterTypes = new[] { if you like. Also could be var method = tippingServiceType.GetInterfaces()....
Both good points. Done. 👍 

> I think I'd still do BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly (REFL008).
Also done. 🙂